### PR TITLE
Fix CI issue SQL Server 2022 failed

### PR DESCRIPTION
**Pull request by kota2and3kan**
## Description

This PR fixes the following CI error.
https://github.com/scalar-labs/scalardb/actions/runs/10073247103/job/27861606398?pr=2075

Recently, the container image of MS SQL Server 2022 was updated, and the file path in the container was changed from `/opt/mssql-tools/bin/sqlcmd` to `/opt/mssql-tools18/bin/sqlcmd`.

- Related issue
  - https://github.com/microsoft/mssql-docker/issues/892
- Official document
  - https://learn.microsoft.com/en-us/sql/linux/quickstart-install-connect-docker
    ```
    In SQL Server 2022 (16.x) CU 14 and later versions, container images include the new mssql-tools18 package. The previous directory /opt/mssql-tools/bin is being phased out. The new directory for Microsoft ODBC 18 tools is /opt/mssql-tools18/bin, aligning with the latest tools offering. For more information about changes and security enhancements, see ODBC Driver 18.0 for SQL Server Released.
    ```

So, I updated the script `create-no-superuser-sqlserver.sh`:

- Check the version of the target SQL Server from the argument.
- Use `mssql-tools18` if the target SQL Server version is 2022.

Also, if we use the new version tool `mssql-tools18` it forces server certificate verification, and it causes the following error.

```console
Sqlcmd: Error: Microsoft ODBC Driver 18 for SQL Server : SSL Provider: [error:0A000086:SSL routines::certificate verify failed:self-signed certificate].
Sqlcmd: Error: Microsoft ODBC Driver 18 for SQL Server : Client unable to establish connection. For solutions related to encryption errors, see https://go.microsoft.com/fwlink/?linkid=2226722.
```

So, I added the `-C` option to implicitly trust the server certificate without validation.
https://learn.microsoft.com/en-us/sql/tools/sqlcmd/sqlcmd-utility

Please take a look!

## Related issues and/or PRs

- https://github.com/scalar-labs/scalardb/actions/runs/10073247103/job/27861606398?pr=2075

## Changes made

- Update `create-no-superuser-sqlserver.sh`.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

You can see the test result in the following URL.

https://github.com/scalar-labs/scalardb/actions/runs/10092203559

## Release notes

N/A

### CODE REVIEW

File: ci/no-superuser/create-no-superuser-sqlserver.sh [hunk 1]
```diff
@@ -21,7 +21,12 @@ do
 
     echo "INFO: Retry count: ${COUNT}"
 
-    docker exec -t ${SQL_SERVER_CONTAINER_NAME} /opt/mssql-tools/bin/sqlcmd -S localhost -U sa -P ${SQL_SERVER_PASSWORD} -d master -Q "SELECT 1"
+    if [[ "$1" = "sqlserver22" ]]; then
```
Comment (kota2and3kan) [inline ci/no-superuser/create-no-superuser-sqlserver.sh:24]: Check the version of the target SQL Server.

File: ci/no-superuser/create-no-superuser-sqlserver.sh [hunk 2]
```diff
@@ -21,7 +21,12 @@ do
...
+        docker exec -t ${SQL_SERVER_CONTAINER_NAME} /opt/mssql-tools18/bin/sqlcmd -S localhost -U sa -P ${SQL_SERVER_PASSWORD} -d master -C -Q "SELECT 1"
```
Comment (kota2and3kan) [inline ci/no-superuser/create-no-superuser-sqlserver.sh:25]: If the target SQL Server is `2022`, we have to use the new command `/opt/mssql-tools18/bin/sqlcmd`.

File: ci/no-superuser/create-no-superuser-sqlserver.sh [hunk 3]
```diff
@@ -21,7 +21,12 @@ do
...
+    else
+        docker exec -t ${SQL_SERVER_CONTAINER_NAME} /opt/mssql-tools/bin/sqlcmd -S localhost -U sa -P ${SQL_SERVER_PASSWORD} -d master -Q "SELECT 1"
```
Comment (kota2and3kan) [inline ci/no-superuser/create-no-superuser-sqlserver.sh:27]: If the target SQL Server is `2019` or `2017`, we can continue to use the current command `/opt/mssql-tools/bin/sqlcmd`.

File: ci/no-superuser/create-no-superuser-sqlserver.sh (hunk 1 above)
Comment (brfrn169) [inline ci/no-superuser/create-no-superuser-sqlserver.sh:24]: For readability, how about extracting the common part into a function like this?

```bash
function sqlcmd () {
    if [[ "$1" = "sqlserver22" ]]; then
        docker exec -t ${SQL_SERVER_CONTAINER_NAME} /opt/mssql-tools18/bin/sqlcmd -S localhost -U $2 -P $3 -d $4 -C -Q $5
    else
        docker exec -t ${SQL_SERVER_CONTAINER_NAME} /opt/mssql-tools/bin/sqlcmd -S localhost -U $2 -P $3 -d $4 -Q $5
    fi
}
```

Comment (komamitsu) [reply ci/no-superuser/create-no-superuser-sqlserver.sh:24]: Another option might be defining a variable for the path to `sqlcmd` above based on the container name?

```shell
if [[ "$SQL_SERVER_CONTAINER_NAME" = "sqlserver22" ]]; then
  SQLCMD=/opt/mssql-tools18/bin/sqlcmd
else
  SQLCMD=/opt/mssql-tools/bin/sqlcmd
fi

...

docker exec -t ${SQL_SERVER_CONTAINER_NAME} ${SQLCMD} -S localhost
```

Comment (kota2and3kan) [reply ci/no-superuser/create-no-superuser-sqlserver.sh:24]: Ah, I see. Looks good!
I will update the script! Thank you for your suggestion!

Comment (kota2and3kan) [reply ci/no-superuser/create-no-superuser-sqlserver.sh]: @brfrn169 @komamitsu 
I fixed it in 218608a6d17495fae60091b7c9699cf7fc90efc4 and 51428be622c36f99545c17831fadab540bc0b523 .
(Sorry, I overlooked @komamitsu 's suggestion. So, I updated it twice.😇 But finally, I adopted `using variables` to switch two versions of `sqlcmd`.)

**Review by Torch3333 — APPROVED** [review]
LGTM, thank you!

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Review by komamitsu — APPROVED** [review]
LGTM, thank you!

**Pull request merged**